tests/gen_cert_test.py

```diff
@@ -181,3 +216,70 @@ def test_cert_date(self, force_date, locale, timezone, expected):
         """
         cert_date = get_cert_date(force_date, locale, timezone)
         assert_equal(expected, cert_date)
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
+    """
+    Tests the subtemplates feature for certs
+
+    Subtemplates are currently written to key on designations only
+
+    These tests have a specific order but do not need to be run in order.
+    They were split and ordered to more easily identify errors while testing.
+    """
+    subtemplate_courses = [
+        'Test/Subtemplates/ByDesignations',
+    ]
+
+    def test_subtemplate_courses_exist(self):
+        """
+        Check that the subtemplate courses exist in settings.CERT_DATA
+        """
+        found_sub_course_ids = []
+        for course_id, course_data in settings.CERT_DATA.iteritems():
+            if course_data.get('subtemplates'):
+                found_sub_course_ids.append(course_id)
+        assert_equal(
+            set(self.subtemplate_courses),
+            set(found_sub_course_ids)
```

Comment:
They are cast to sets because then element order in lists doesn't matter.  Lists with the same elements in different order are not equal.